src/promptflow/promptflow/_sdk/_service/apis/collector.py

```diff
@@ -29,14 +29,21 @@
 from promptflow._utils.thread_utils import ThreadWithContextVars
 
 
-def trace_collector(get_created_by_info_with_cache: Callable, logger: logging.Logger):
+def trace_collector(
+    get_created_by_info_with_cache: Callable,
+    logger: logging.Logger,
+    cloud_trace_only: bool = False,
+    credential: object = None,
```

Comment:
```suggestion
    credential: typing.Optional[object] = None,
```

Reply:
`azure.core.credentials.TokenCredential` might be a better typing?

Reply:
Do we need to avoid the import of "azure.core.credentials.TokenCredential" here?